Automattic/simplenote-macos
# Scrollable Tags Editor

### Details:
In this PR we're replicating iOS's Tags Editor behavior:

1. From now on, the Tags Editor will live below the Note Editor Area
2. Whenever the document is short, the Editor will be anchored at the bottom edge of the screen
3. Whenever the document is long enough, the Editor will scroll along with the Note's contents,
 
Closes #985

@eshurakov Sir! May I bug you with this one as well?
**Thanks in advance!!**

### Test: New Note
1. Add a new Note

- [x] Verify the Top Padding of the first line of text (Editor) matches the Notes List's first cell
- [x] Verify the Tags Editor is anchored at the bottom
- [x] Verify that you can click over the Tags Editor
- [x] Verify that if you add enough content, the Tags Editor scrolls along horizontally
- [x] Verify that you can remove or edit Tags

### Test: Existing Short Note
1. Click over any super short note

- [x] Verify the Tags Editor is anchored at the bottom of the Editor

### Test: Existing Long Note
1. Click over any super long note

- [x] Verify the Tags Editor is not immediately visible
- [x] Verify that if you scroll down, the Editor is revealed

### Test: Markdown Preview
1. Enable Markdown Preview in a note
2. While in Edition mode, click over the Tags Editor
3. Press the Markdown Preview button

- [x] Verify you can no longer edit the Tags (while in MD preview!)

### Test: UX
- [x] Verify the Tags Editor placeholder reads as `Tag...` (below the content editor!)
- [x] Verify that in a note with tons of Tags, the Tags Editor no longer clips its contents. [Ref. this comment!](https://github.com/Automattic/simplenote-macos/pull/987#issuecomment-872295078)
- [x] Verify that in a note with lots of text, pressing **CMD + Shift + Y** causes the Editor to scroll all the way down (and thus, making the Tags Editor visible)

### Release
`RELEASE-NOTES.txt` was updated in 7c31128 with:

> Tags Editor is now scrollable, making more room for the editor when appropriate
